ui/StatusQ/src/keychain.mm

```diff
@@ -0,0 +1,262 @@
+#include "StatusQ/keychain.h"
+
+#include <QDebug>
+#include <QEventLoop>
+#include <QFuture>
+#include <QFutureWatcher>
+#include <QtConcurrent/QtConcurrent>
+
+#include <Foundation/Foundation.h>
+#include <LocalAuthentication/LocalAuthentication.h>
+#include <Security/Security.h>
+
+const static auto authPolicy =
+#if defined(MAC_OS_VERSION_15_0) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_VERSION_15_0
+    LAPolicyDeviceOwnerAuthenticationWithBiometricsOrCompanion;
+#elif defined(MAC_OS_X_VERSION_10_15) && MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_15
+    LAPolicyDeviceOwnerAuthenticationWithBiometricsOrWatch;
+#else
+    LAPolicyDeviceOwnerAuthenticationWithBiometrics
+#endif
+
+static Keychain::Status convertStatus(OSStatus status)
+{
+    switch (status) {
+    case errSecSuccess:
+        return Keychain::StatusSuccess;
+    case errSecItemNotFound:
+        return Keychain::StatusNotFound;
+    case errSecCSCancelled:
+        return Keychain::StatusCancelled;
+    default:
+        return Keychain::StatusGenericError;
+    }
+}
+
+Keychain::Status convertError(NSError *error)
+{
+    switch (error.code) {
+    case errSecSuccess:
+        return Keychain::StatusSuccess;
+    case LAErrorSystemCancel:
+    case LAErrorUserCancel:
+    case LAErrorAppCancel:
+        return Keychain::StatusCancelled;
+    default:
+        return Keychain::StatusGenericError;
+    }
+}
+
+Keychain::~Keychain()
+{
+    QFutureWatcher<void> watcher;
+    cancelActiveRequest();
+    watcher.setFuture(m_future);
+    watcher.waitForFinished();
```

Comment:
```suggestion
    cancelActiveRequest();
    m_future.waitForFinished();
```
Cannot we call `waitForFinished` on the `QFuture` directly?

Reply:
indeed!